src/features/serde/de_owned.rs

```diff
@@ -24,28 +24,37 @@ impl<DE: Decoder> OwnedSerdeDecoder<DE> {
 }
 
 #[cfg(feature = "std")]
-impl<'r, C: Config, R: std::io::Read> OwnedSerdeDecoder<DecoderImpl<IoReader<&'r mut R>, C>> {
+impl<'r, 'context, C: Config, R: std::io::Read, Context>
+    OwnedSerdeDecoder<DecoderImpl<'context, IoReader<&'r mut R>, C, Context>>
+{
     /// Creates the decoder from an `std::io::Read` implementor.
     pub fn from_std_read(
         src: &'r mut R,
         config: C,
-    ) -> OwnedSerdeDecoder<DecoderImpl<IoReader<&'r mut R>, C>>
+        context: &'context mut Context,
+    ) -> OwnedSerdeDecoder<DecoderImpl<'context, IoReader<&'r mut R>, C, Context>>
     where
         C: Config,
     {
         let reader = IoReader::new(src);
-        let decoder = DecoderImpl::new(reader, config);
+        let decoder = DecoderImpl::new(reader, config, context);
         Self { de: decoder }
     }
 }
 
-impl<C: Config, R: Reader> OwnedSerdeDecoder<DecoderImpl<R, C>> {
+impl<'context, C: Config, R: Reader, Context>
+    OwnedSerdeDecoder<DecoderImpl<'context, R, C, Context>>
+{
     /// Creates the decoder from a [`Reader`] implementor.
-    pub fn from_reader(reader: R, config: C) -> OwnedSerdeDecoder<DecoderImpl<R, C>>
+    pub fn from_reader(
+        reader: R,
+        config: C,
+        context: &'context mut Context,
+    ) -> OwnedSerdeDecoder<DecoderImpl<'context, R, C, Context>>
```

Comment:
Might be nice to leave this function context-less and add a new `from_reader_with_context` function